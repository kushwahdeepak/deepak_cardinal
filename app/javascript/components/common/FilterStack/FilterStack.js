import feather from 'feather-icons'
import { nanoid } from 'nanoid'
import _ from 'lodash'

import Expander from '../../common/Expander/Expander'
import TagInput from '../../common/inputs/TagInput/TagInput'
import MultiBooleanInput from '../../common/inputs/MultiBooleanInput/MultiBooleanInput'
import InputSection from '../../common/inputs/InputSection/InputSection'
import BooleanInput from '../inputs/BooleanInput/BooleanInput'
import RadioInput from '../inputs/RadioInput/RadioInput'
import FilterGroup from '../../common/FilterGroup/FilterGroup'
import React, { useState, useEffect } from 'react'
import Button from 'react-bootstrap/Button'
import isNil from 'lodash.isnil'
import './style/FilterStack.scss'

const FilterStack = ({ filterStack, currentOrganization, setFilterData, memberOrganization, handleSearch, candidatePage, setFilterStack, emptyFilter, closeFunc, jobId}) => {
    const userBelongsToCT = currentOrganization?.name === window.ch_const.ct_org_name

    const subOrgBelongsToCT = memberOrganization?.name === window.ch_const.ct_org_name

    useEffect(() => {
        feather.replace()
    })
    const [checked, setChecked] = useState(emptyFilter)
    const manualSearch = !isNil(handleSearch)

    const handleSubmit = (e) => {
        e.preventDefault()
        handleSearch()
    }

    const resetFilterStack = (e) => {
      e.preventDefault();
      const filter = {...emptyFilter}
      filter.keyword = filterStack.keyword
      setChecked(filter)
      setFilterStack(filter)
      closeFunc()
    }

    const setCheckBox = (attr,val) => {
        const newBoolStack = { ...checked, [attr]: val }
     if(!_.isEqual(checked,newBoolStack)) {
        setChecked(newBoolStack)
     }
    }

    function activeFilterCount(filters) {
        return Object.keys(filters)
            .map((key) => {
                if (Array.isArray(filterStack[key]))
                    return filterStack[key].length > 0 ? 1 : 0
                else if (typeof filterStack[key] === 'boolean')
                    return filterStack[key] ? 1 : 0
                else return 0
            })
            .reduce((acc, current) => acc + current, 0)
    }

    return (

        <FilterGroup candidatePage={candidatePage} filterCount={activeFilterCount(filterStack)}>

            <Expander
                label="Recruiter Activity"
                expanded={
                    filterStack.topSchool ||
                    filterStack.topCompany ||
                    filterStack.active
                }
                candidatePage={candidatePage}
            >
                <InputSection>
                    <BooleanInput
                        setInputState={setFilterData}
                        label="Top school"
                        testAttr="top_school"
                        initialValue={filterStack.top_school}
                        checked={checked.top_school}
                        setCheckBox={setCheckBox}
                    />
                    <BooleanInput
                        setInputState={setFilterData}
                        label="Top company"
                        testAttr="top_company"
                        initialValue={filterStack.top_company}
                        checked={checked.top_company}
                        setCheckBox={setCheckBox}
                    />
                    { (userBelongsToCT || subOrgBelongsToCT) && 
                        <BooleanInput
                            setInputState={setFilterData}
                            label="Active"
                            testAttr="active"
                            initialValue={filterStack.active}
                            checked={checked.active}
                            setCheckBox={setCheckBox}
                        />
                    }
                </InputSection>
            </Expander>
            <Expander
                label="Company"
                expanded={
                    filterStack.companyNames &&
                    filterStack.companyNames.length > 0
                }
                candidatePage={candidatePage}
            >
                <InputSection>
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.company_names
                                ? Array.isArray(filterStack.company_names)
                                    ? filterStack.company_names
                                          .filter((company) => company != '')
                                          .map((company) => {
                                              return {
                                                  id: nanoid(),
                                                  value: company,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="company_names"
                        candidatePage={candidatePage}
                    />
                </InputSection>
            </Expander>
            <Expander
                label="Education"
                expanded={
                    (filterStack.schools && filterStack.schools.length > 0) ||
                    (filterStack.disciplines &&
                        filterStack.disciplines.length > 0) ||
                    (filterStack.degrees && filterStack.degrees.length > 0)
                }
                candidatePage={candidatePage}

            >
                <InputSection label="School">
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.schools
                                ? Array.isArray(filterStack.schools)
                                    ? filterStack.schools
                                          .filter((school) => school != '')
                                          .map((school) => {
                                              return {
                                                  id: nanoid(),
                                                  value: school,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="schools"
                        candidatePage={candidatePage}
                    />
                </InputSection>
                {!jobId && location.pathname != "/people_searches/new" ? 
                <InputSection label="Discipline (major)">
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.disciplines
                                ? Array.isArray(filterStack.disciplines)
                                    ? filterStack.disciplines
                                          .filter(
                                              (discipline) => discipline != ''
                                          )
                                          .map((discipline) => {
                                              return {
                                                  id: nanoid(),
                                                  value: discipline,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="disciplines"
                        candidatePage={candidatePage}
                    />
                </InputSection>: ""}
                {!jobId ? 
                <InputSection label="Degrees">
                    <MultiBooleanInput
                        setInputState={setFilterData}
                        label="Degrees"
                        checkboxInfo={[
                            {
                                label: window.ch_const.degree_bachelors,
                                option: window.ch_const.degree_bachelors,
                            },
                            {
                                label: window.ch_const.degree_masters,
                                option: window.ch_const.degree_masters,
                            },
                            {
                                label: window.ch_const.degree_doctorate,
                                option: window.ch_const.degree_doctorate,
                            },
                        ]}
                        initialValues={filterStack.degrees}
                        testAttr="degrees"
                        candidatePage={candidatePage}
                    />
                </InputSection> : ""}
            </Expander>
            <Expander
                label="Location"
                expanded={
                    filterStack.locations && filterStack.locations.length > 0
                }
                candidatePage={candidatePage}
            >
                <InputSection>
                  <div>{"State"}</div>
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.state
                                ? Array.isArray(filterStack.state)
                                    ? filterStack.state
                                          .filter((stateName) => stateName != '')
                                          .map((stateName) => {
                                              return {
                                                  id: nanoid(),
                                                  value: stateName,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="state"
                        candidatePage={candidatePage}
                        showTag={filterStack?.city && filterStack?.city.length == 0 ? true : false}
                    />
                    <div>{"City"}</div>
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.city
                                ? Array.isArray(filterStack.city)
                                    ? filterStack.city
                                          .filter((cityName) => cityName != '')
                                          .map((cityName) => {
                                              return {
                                                  id: nanoid(),
                                                  value: cityName,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="city"
                        candidatePage={candidatePage}
                        showTag={filterStack?.state && filterStack?.state.length == 0 ? true : false}
                    />
                </InputSection>
            </Expander>
            <Expander
              label="Skills"
              expanded={filterStack.skills && filterStack.skills.length > 0}
              candidatePage={candidatePage}
            >
              <InputSection>
                <TagInput
                  setInputState={setFilterData}
                  initialValues={
                    filterStack.skills
                      ? Array.isArray(filterStack.skills)
                        ? filterStack.skills
                            .filter((skill) => skill != '')
                            .map((skill) => {
                              return {
                                id: nanoid(),
                                value: skill,
                              }
                            })
                        : []
                      : []
                  }
                  testAttr="skills"
                  candidatePage={candidatePage}
                />
              </InputSection>
            </Expander>
            <Expander
                label="Job Title"
                expanded={
                    filterStack.titles && filterStack.titles.length > 0
                }
                candidatePage={candidatePage}
            >
                <InputSection>
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.titles
                                ? Array.isArray(filterStack.titles)
                                    ? filterStack.titles
                                          .filter((titles) => titles != '')
                                          .map((titles) => {
                                              return {
                                                  id: nanoid(),
                                                  value: titles,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="titles"
                        candidatePage={candidatePage}
                    />
                </InputSection>
            </Expander>
            <Expander
                label="Phone number"
                expanded={filterStack.phoneNumberAvailable}

                candidatePage={candidatePage}
            >
                <InputSection>
                    <BooleanInput
                        setInputState={setFilterData}
                        label="Phone number available"
                        testAttr="phone_number_available"
                        initialValue={filterStack.phone_number_available}
                        checked={checked.phone_number_available}
                        setCheckBox={setCheckBox}
                    />
                </InputSection>
            </Expander>
            <Expander
                label="Emails"
                expanded={filterStack.emails && filterStack.emails.length > 0}

                candidatePage={candidatePage}

            >
                <InputSection>
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.emails
                                ? Array.isArray(filterStack.emails)
                                    ? filterStack.emails
                                          .filter((email) => email != '')
                                          .map((email) => {
                                              return {
                                                  id: nanoid(),
                                                  value: email,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="emails"
                        candidatePage={candidatePage}
                    />
                </InputSection>
                <RadioInput
                    setInputState={setFilterData}
                    radioInfo={['Personal', 'Company']}
                    initialValue={filterStack.email_type}
                    testAttr="email_type"
                />
            </Expander>
            <Expander
                label="Names"
                expanded={filterStack.names && filterStack.names.length > 0}

                candidatePage={candidatePage}
            >
                <InputSection>
                    <TagInput
                        setInputState={setFilterData}
                        initialValues={
                            filterStack.names
                                ? Array.isArray(filterStack.names)
                                    ? filterStack.names
                                          .filter((name) => name != '')
                                          .map((name) => {
                                              return {
                                                  id: nanoid(),
                                                  value: name,
                                              }
                                          })
                                    : []
                                : []
                        }
                        testAttr="names"
                        candidatePage={candidatePage}
                    />
                </InputSection>
            </Expander>
            <button className='clear-filter' onClick={(e) => resetFilterStack(e)}>  Clear Filters <i className='icon-sm' data-feather='x' /></button>
        </FilterGroup>
    )
}

export default FilterStack
