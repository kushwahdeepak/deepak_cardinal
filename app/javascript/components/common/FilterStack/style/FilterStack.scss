.clear-filter{ 
  background-color: white;
  text-align: center;
  font-size: 14px;
  margin-top: 15px;
  color: #474953;
}

.text-font {
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 19px;
}
