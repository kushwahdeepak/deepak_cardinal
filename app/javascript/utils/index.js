import { act } from 'react-dom/test-utils'

export const findByTestAttr = (wrapper, attr) => {
    return wrapper.find(`[data-test='${attr}']`)
}

export const timeoutResolve = (time = 0, response = {}) =>
    new Promise((resolve) => {
        setTimeout(() => resolve(response), time)
    })

export const wait = (time = 0) => {
    const timer = timeoutResolve(time)

    return act(async () => {
        await timer
    })
}
export function emailIsValid(email) {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
}

export function generateRandom(min, max) {
    return Math.floor(min + Math.random() * (max + 1 - min))
}

export function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
}

export function createMarkedList(array) {
    if (!Array.isArray(array)) {
        return []
    }
    return array.map((candidate) => ({
        id: candidate.id,
        name: candidate.id,
        fullName: `${candidate.first_name} ${candidate.last_name}`,
        emailAddress: candidate.email_address,
        isChecked: false,
    }))
}

export function stringReplace(string,toSymbol,fromSymbol){
    return string.replace(toSymbol, fromSymbol)
}

export function capitalize(string) {
    return string.replace(/\b(\w)/g, s => s.toUpperCase());
}

export function skillIsNotEmpty(value){
    return value.trim().length > 0;
}

export function firstCharacterCapital(value){
    return value?.charAt(0)?.toUpperCase() + value?.slice(1)
}
