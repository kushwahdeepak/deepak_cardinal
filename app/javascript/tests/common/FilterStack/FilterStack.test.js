import React from 'react';
import FilterStack from '../../../components/common/FilterStack/FilterStack';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { findByTestAttr } from '../../../utils';

configure({adapter: new Adapter()});

const initialShallowRender = (props = {}) => {
	return shallow(<FilterStack {...props}/>);
};

const initialMountRender = (props = {}) => {
	return mount(<FilterStack {...props}/>);
};

window.ch_const = {};
const initialProps = {
	filterStack: {},
	setStackFilter: jest.fn()
};


describe('FilterStack component',  () => {
	let wrapper;
	let fullWrapper;

	beforeEach(() => {

		const props = {
			...initialProps
		};
		wrapper = initialShallowRender(props);
		fullWrapper = initialMountRender(props);
	});

	afterEach(() => {
		fullWrapper && fullWrapper.unmount();
	});

	it('should render FilterStack component', () => {
		expect(wrapper.exists()).toBe(true);
	});

	it('should match with a snapshot', () => {
		const component = renderer.create(<FilterStack {...initialProps} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('FilterStack component with props', () => {

	});
	describe('FilterStack component without props', () => {

	});
});
