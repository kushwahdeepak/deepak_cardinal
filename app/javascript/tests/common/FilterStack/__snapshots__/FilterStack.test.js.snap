// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterStack component should match with a snapshot 1`] = `
<div
  className="undefined shadow-sm card"
>
  <div
    className="p-2 d-flex justify-content-between card-header"
    data-test="filters-card-header"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.35)",
      }
    }
  >
    <button
      className="btn btn-link"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="text-danger"
        data-feather="trash-2"
      />
    </button>
  </div>
  <div
    className="p-2 card-body"
  >
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Highlights
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="d-flex align-items-center"
            >
              <input
                checked={false}
                data-test="checkbox-item"
                id="topSchool"
                onChange={[Function]}
                style={
                  Object {
                    "marginRight": "0.5rem",
                  }
                }
                type="checkbox"
              />
              <label
                data-test="label-item"
                htmlFor="topSchool"
                style={
                  Object {
                    "marginBottom": "0",
                  }
                }
              >
                Top school
              </label>
            </div>
            <div
              className="d-flex align-items-center"
            >
              <input
                checked={false}
                data-test="checkbox-item"
                id="topCompany"
                onChange={[Function]}
                style={
                  Object {
                    "marginRight": "0.5rem",
                  }
                }
                type="checkbox"
              />
              <label
                data-test="label-item"
                htmlFor="topCompany"
                style={
                  Object {
                    "marginBottom": "0",
                  }
                }
              >
                Top company
              </label>
            </div>
            <div
              className="d-flex align-items-center"
            >
              <input
                checked={false}
                data-test="checkbox-item"
                id="active"
                onChange={[Function]}
                style={
                  Object {
                    "marginRight": "0.5rem",
                  }
                }
                type="checkbox"
              />
              <label
                data-test="label-item"
                htmlFor="active"
                style={
                  Object {
                    "marginBottom": "0",
                  }
                }
              >
                Active
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Company
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Education
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          >
            School
          </div>
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            className="med-light-label"
          >
            Discipline (major)
          </div>
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            className="med-light-label"
          >
            Degrees
          </div>
          <div
            className="p-2"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <div
                    className="label-and-checkbox"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      style={
                        Object {
                          "marginRight": "0.5rem",
                        }
                      }
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="label-and-checkbox"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      style={
                        Object {
                          "marginRight": "0.5rem",
                        }
                      }
                      type="checkbox"
                    />
                  </div>
                  <div
                    className="label-and-checkbox"
                  >
                    <input
                      checked={false}
                      onChange={[Function]}
                      style={
                        Object {
                          "marginRight": "0.5rem",
                        }
                      }
                      type="checkbox"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Location
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Skills
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Tags
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Job Title
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Phone number
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="d-flex align-items-center"
            >
              <input
                checked={false}
                data-test="checkbox-item"
                id="phoneNumberAvailable"
                onChange={[Function]}
                style={
                  Object {
                    "marginRight": "0.5rem",
                  }
                }
                type="checkbox"
              />
              <label
                data-test="label-item"
                htmlFor="phoneNumberAvailable"
                style={
                  Object {
                    "marginBottom": "0",
                  }
                }
              >
                Phone number available
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Emails
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="accordion"
      data-test="accordion-item"
    >
      <button
        className="btn btn-expander"
        data-test="expander-button"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "none",
            "fontWeight": 500,
            "width": "100%",
          }
        }
        type="Button"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          Names
          <div
            style={
              Object {
                "display": "block",
              }
            }
          >
            <i
              data-feather="chevron-down"
            />
          </div>
          <div
            style={
              Object {
                "display": "none",
              }
            }
          >
            <i
              data-feather="chevron-right"
            />
          </div>
        </div>
      </button>
      <div
        aria-expanded={null}
        className="collapse"
      >
        <div
          className="container"
          style={
            Object {
              "paddingBottom": "0",
            }
          }
        >
          <div
            className="med-light-label"
          />
          <div
            className="p-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-white"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="css-i6dzq1"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 5L12 19"
                  />
                  <path
                    d="M5 12L19 12"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Filter
    </button>
  </div>
</div>
`;
